### ISSUE TITLE
SqlNoDeserializationTest doesn't work with Jet engine

### ISSUE THREAD

**Issue by Fly-Style** [hazelcast/hazelcast#19273]

**Comment by Fly-Style** [comment]
Failed again on fresh master, but with new cause

<details>
```
com.hazelcast.sql.impl.QueryException: Failed to deserialize map entry key or value: java.io.IOException: KEY FAILURE
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:65) ~[classes/:?]
	at com.hazelcast.sql.impl.QueryException.dataException(QueryException.java:98) ~[classes/:?]
	at com.hazelcast.sql.impl.LazyTarget.deserialize(LazyTarget.java:63) ~[classes/:?]
	at com.hazelcast.sql.impl.expression.ColumnExpression.unwrapLazyValue(ColumnExpression.java:80) ~[classes/:?]
	at com.hazelcast.sql.impl.expression.ColumnExpression.eval(ColumnExpression.java:71) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate(ExpressionUtil.java:211) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.keyvalue.KvRowProjector.project(KvRowProjector.java:116) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.keyvalue.KvRowProjector.project(KvRowProjector.java:103) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.map.RowProjectorProcessorSupplier.lambda$get$1c00851f$1(RowProjectorProcessorSupplier.java:71) ~[classes/:?]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:48) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor$FlatMapper.tryProcess(AbstractProcessor.java:566) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.TransformP.tryProcess(TransformP.java:45) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:590) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:108) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$processInbox$2f647568$2(ProcessorTasklet.java:440) ~[classes/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:506) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:440) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:305) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:300) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:281) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:255) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:388) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) [?:1.8.0_312]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:356) [classes/:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_312]
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: java.io.IOException: KEY FAILURE
	at com.hazelcast.internal.serialization.impl.SerializationUtil.handleException(SerializationUtil.java:111) ~[classes/:?]
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:276) ~[classes/:?]
	at com.hazelcast.sql.impl.LazyTarget.deserialize(LazyTarget.java:56) ~[classes/:?]
	... 24 more
Caused by: java.io.IOException: KEY FAILURE
	at com.hazelcast.jet.sql.impl.misc.SqlNoDeserializationTest$PersonKey.readPortable(SqlNoDeserializationTest.java:237) ~[test-classes/:?]
	at com.hazelcast.internal.serialization.impl.portable.PortableSerializer.read(PortableSerializer.java:305) ~[classes/:?]
	at com.hazelcast.internal.serialization.impl.portable.PortableSerializer.read(PortableSerializer.java:97) ~[classes/:?]
	at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:44) ~[classes/:?]
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:270) ~[classes/:?]
	at com.hazelcast.sql.impl.LazyTarget.deserialize(LazyTarget.java:56) ~[classes/:?]
	... 24 more
```
</details>

**Comment by viliam-durina** [comment]
Will be fixed by #20420

**Comment by viliam-durina** [comment]
Fixed by #20420
